### PR TITLE
CI: Fix CI failures on master

### PR DESCRIPTION
**Pull request by TomAugspurger**
(WIP for now). Starting to track them down.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
One is related to https://github.com/dask/dask/pull/2310/

If a user passes `Path("s3://bucket/key.csv")`, but doesn't have s3fs installed, we raise an ImportError.  So they install s3fs, re-run it, and then we raise an OSError. Not ideal, but I think that's OK. Then the fix is to skip that test when s3fs isn't installed.

**Comment by TomAugspurger** [comment]
https://travis-ci.org/dask/dask/jobs/243250337 should be fixed by adding `--doctest-ignore-import-errors` to the run doing doctests. Not sure why it didn't fail before though.

**Comment by TomAugspurger** [comment]
Though we probably want all the optional dependencies installed for the doctest run.

**Comment by TomAugspurger** [comment]
Seems like s3fs wasn't being installed anywhere on Travis. My last commit installs it always, though perhaps we want a build without it, to ensure we never accidentally require it to import dask.

**Comment by TomAugspurger** [comment]
I'm able to reproduce the hang (like in https://travis-ci.org/dask/dask/jobs/243394188) locally. Here's the traceback with I terminate it

```python
^CProcess ForkPoolWorker-149:
Process ForkPoolWorker-151:
Process ForkPoolWorker-148:
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 108, in worker
    task = get()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/queues.py", line 343, in get
    res = self._reader.recv_bytes()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/connection.py", line 216, in recv_bytes
    buf = self._recv_bytes(maxlength)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/connection.py", line 407, in _recv_bytes
    buf = self._recv(4)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/connection.py", line 379, in _recv
    chunk = read(handle, remaining)
KeyboardInterrupt
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 108, in worker
    task = get()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/queues.py", line 342, in get
    with self._rlock:
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/synchronize.py", line 96, in __enter__
    return self._semlock.__enter__()
KeyboardInterrupt
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 108, in worker
    task = get()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/queues.py", line 342, in get
    with self._rlock:
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/synchronize.py", line 96, in __enter__
    return self._semlock.__enter__()
KeyboardInterrupt
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 108, in worker
    task = get()
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/queues.py", line 342, in get
    with self._rlock:
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/synchronize.py", line 96, in __enter__
    return self._semlock.__enter__()
KeyboardInterrupt
```

**Comment by martindurant** [comment]
The failures in https://github.com/dask/dask/pull/2466 are the same issue?

**Comment by TomAugspurger** [comment]
> The failures in #2466 are the same issue?

I'm still trying to pin down which tests are causing the hang with pytest-xdist. We could simply disable it for now, which will cause everything to take longer (but at least it'll finish). Let me try that quick.

**Comment by jcrist** [comment]
> though perhaps we want a build without it, to ensure we never accidentally require it to import dask.

We have import tests to check the minimal dependencies are actually minimal: https://github.com/dask/dask/blob/master/continuous_integration/travis/test_imports.sh. I'd say we should install everything needed to run all the tests.

File: continuous_integration/travis/install.sh [hunk 1]
```diff
@@ -62,6 +63,7 @@ pip install -q git+https://github.com/dask/partd --upgrade --no-deps
 pip install -q git+https://github.com/dask/zict --upgrade --no-deps
 pip install -q git+https://github.com/dask/distributed --upgrade --no-deps
 pip install -q git+https://github.com/mrocklin/sparse --upgrade --no-deps
+pip install -q git+https://github.com/mrocklin/s3fs --upgrade --no-deps
```
Comment (jcrist) [inline continuous_integration/travis/install.sh]: `mrocklin` -> `dask`

File: continuous_integration/travis/run_tests.sh [hunk 2]
```diff
@@ -1,12 +1,14 @@
 # Need to make test order deterministic when parallelizing tests, hence PYTHONHASHSEED
 # (see https://github.com/pytest-dev/pytest-xdist/issues/63)
 if [[ $PARALLEL == 'true' ]]; then
-    export XTRATESTARGS="-n3 $XTRATESTARGS"
+    # export XTRATESTARGS="-n3 $XTRATESTARGS"
```
Comment (jcrist) [inline continuous_integration/travis/run_tests.sh]: Just a friendly reminder to revert this before merging.

File: dask/bag/tests/test_bag.py [hunk 3]
```diff
@@ -602,6 +602,7 @@ def test_can_use_dict_to_make_concrete():
 
 @pytest.mark.slow
 @pytest.mark.network
+@pytest.mark.skip(reason="Hangs")
```
Comment (TomAugspurger) [inline dask/bag/tests/test_bag.py]: I'm not sure why, but when I ran `pytest` locally with `-n 2`, one of the works got stuck on this test.  The test suite finishes when I skip it locally, we'll see about Travis.

**Comment by TomAugspurger** [comment]
All green. I'll spend a bit more time to figure out *why* those tests are hanging. But I think we should merge this today so that our CI is running again, and open follow up issues for those tests.

**Comment by jcrist** [comment]
Fine by me, thanks for taking care of this.

**Comment by TomAugspurger** [comment]
https://github.com/dask/dask/issues/2471 for the two new skips.

**Pull request merged**